File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/lobby/ListTableCommand.java
package bluffinmuffin.protocol.commands2.lobby;

import java.util.ArrayList;
import java.util.StringTokenizer;

import bluffinmuffin.protocol.TupleTableInfo;

public class ListTableCommand extends AbstractLobbyCommand
{
    @Override
    protected String getCommandName()
    {
        return ListTableCommand.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyLIST_TABLES";
    
    public ListTableCommand(StringTokenizer argsToken)
    {
    }
    
    public ListTableCommand()
    {
    }
    
    public String encodeResponse(ArrayList<TupleTableInfo> tables)
    {
        return new ListTableResponse(this, tables).encode();
    }
}

File: C#/Com.Ericmas001.Game.Poker.Protocol.Commands/Lobby/ListTableCommand.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;
using Com.Ericmas001.Game.Poker.Protocol.Commands.Lobby.Training;
using Com.Ericmas001.Game.Poker.Protocol.Commands.Lobby.Career;
using Com.Ericmas001.Game.Poker.Protocol.Commands.Entities;
using Newtonsoft.Json.Linq;
using Com.Ericmas001.Game.Poker.DataTypes.Enums;

namespace Com.Ericmas001.Game.Poker.Protocol.Commands.Lobby
{
    public class ListTableCommand : AbstractLobbyCommand
    {
        public static string COMMAND_NAME = "lobbyLIST_TABLES";
        public LobbyTypeEnum[] LobbyTypes { get; set; }

        public string EncodeResponse(List<TableInfo> tables)
        {
            return new ListTableResponse(this) { Tables = tables }.Encode();
        }
    }
}
